src/main/java/com/salesforce/emp/connector/EmpConnector.java

```diff
@@ -214,6 +216,27 @@ public void stop() {
         return subscribe(topic, REPLAY_FROM_TIP, consumer);
     }
 
+    public EmpConnector addListener(String channel, ClientSessionChannel.MessageListener messageListener) {
+        listenerInfos.add(new MessageListenerInfo(channel, messageListener));
+        return this;
+    }
+
+    public boolean isConnected() {
+        return client != null && client.isConnected();
+    }
+
+    public boolean isDisconnected() {
+        return client != null && client.isDisconnected();
```

Comment:
Here you should have the logic: `return client == null || client.isDisconnected();`

Reply:
Good catch will change that. Thanks.